server/src/lib/model/user-client.ts

```diff
@@ -262,43 +261,25 @@ const UserClient = {
     invite: string
   ): Promise<boolean> {
     if (email && challenge && team) {
-      let client_id = await Promise.all([UserClient.findClientId(email)]);
+      const client_id = (await Promise.all([
+        UserClient.findClientId(email),
+      ]))[0];
       if (client_id) {
-        // Generate a random, anonymized, and unique invite ID for enrollments.
-        // HACK(riley/can): Can we move this entirely into the DB?
-        let inviteToken;
-        while (true) {
-          inviteToken = (await hash(`${Math.random()}`)).slice(0, 12);
-          const res = await db.query(
-            `SELECT * FROM enrollments WHERE url_token=?`,
-            inviteToken
-          );
-          if (!res || !res.length) break;
-        }
-
+        // if invite is null, the invitation comes from team; OTHERWISE from user
+        // for user invitation, invite_by is the same value of url_token from another row
+        // for team invitation, invite_by is the same value of url_token of the same row
+        const [[{ team_id, challenge_id, uuid }]] = await db.query(
+          `SELECT t.id AS team_id, t.challenge_id, UUID() AS uuid FROM teams t WHERE t.url_token=?`,
```

Comment:
I still think `UUID()` output is pretty long for a public invite ID. Are you sure we can't use `UUID_SHORT()`?

An easy win is `REPLACE(UUID(),'-','')`. If we have time, we can save another 10 characters by base 58 encoding the strings on the server before sending them down, eg:

```
const base58Alphabet = '123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ';

function encodeInviteId(uuid, chars) {
	const base = chars.length;
	let decimalUuid = parseInt(uuid, 16);
	let inviteId = '';
	while (decimalUuid) {
		inviteId += chars[decimalUuid % base];
		decimalUuid = Math.floor(decimalUuid / base);
	}
	return inviteId.padStart(22, chars[0]);
}

function decodeInviteId(inviteId, chars) {
	// ...
}

// Example usage: encodeInviteId(uuid, base58Alphabet);
```

Let's not prioritize this. It's a nice to have but not on the critical path.

Reply:
MySQL have conv() that can do the base encoding stuff, but mysql can only base 32 encode the string. 

Beside, parseInt() in js won't support numbers as big as UUID which is 128 bits long, 53 bits at most, then the precision is lost. 

Remove '-' from UUID, and base 64 encoding the string (plus 0,- and ~) will save 1/3 of the total length. I think the performance won't change much for that. It is a great idea, but I think it is better to leave it for future. I can leave a comments like `[FUTURE]`

Reply:
As for `UUID_SHORT()`, run `select uuid_short()` any times in a mysql shell, different values will be yield. However, enroll two users, they will get the same short uuid, thus creating a _duplicate key exception_, and failing the second enrollment.

That is what I observed in my local dev environment (win-10, mysql-5.6.40 and mysql 8.0.17), not sure if it is the same on your environment.

Reply:
Thanks for the clarification, and for adding the `[FUTURE]` comment! I'm happy with this solution for now.